tests/integration.rs

```diff
@@ -27,8 +27,7 @@ fn result_succeeding() {
 
     fn test() -> Result<(), Error> {
         // make sure 'ls' is installed
-        let result: Result<(), Error> = cmd!(WHICH, "ls");
-        result?;
+        cmd_result!(WHICH, "ls")?;
```

Comment:
I'm really surprised that this works without type annotations. I thought it would have to be e.g. `let () = cmd_result!(WHICH, "ls")?;`. There seems to be some type defaulting to be going on.

(cc: @casey)

Reply:
Here's some explanation of the rustc behavior: https://www.reddit.com/r/rust/comments/f5nw1o/why_does_rust_infer_when_the_generic_type_cannot/

Reply:
That's interesting, I guess `()` is just a default in some circumstances.